lib/backend/k8s/resources/customresource.go

```diff
@@ -310,3 +326,63 @@ func (c *customK8sResourceClient) Watch(ctx context.Context, list model.ListInte
 func (c *customK8sResourceClient) EnsureInitialized() error {
 	return nil
 }
+
+func listInterfaceToKey(l model.ListInterface) model.Key {
+	pl := l.(model.ResourceListOptions)
+	if pl.Name != "" {
+		return model.ResourceKey{Name: pl.Name, Kind: pl.Kind}
+	}
+	return nil
+}
+
+func keyToName(k model.Key) (string, error) {
+	return k.(model.ResourceKey).Name, nil
+}
+
+func convertResourceToKVPair(r Resource, resourceKind string) (*model.KVPair, error) {
+	kvp := &model.KVPair{
+		Key: model.ResourceKey{
+			Name:      r.GetObjectMeta().GetName(),
+			Namespace: r.GetObjectMeta().GetNamespace(),
+			Kind:      resourceKind,
+		},
+		Revision: r.GetObjectMeta().GetResourceVersion(),
+	}
+
+	var v2resource Resource
+	switch resourceKind {
+	case apiv2.KindBGPPeer:
+		val := apiv2.NewBGPPeer()
+		val.Spec = r.(*apiv2.BGPPeer).Spec
+		v2resource = val
+	default:
+		// TODO: make this a better error msg
+		return nil, errors.New("Invalid resource type")
+	}
+
+	v2resource.GetObjectMeta().SetName(r.GetObjectMeta().GetName())
+	v2resource.GetObjectMeta().SetNamespace(r.GetObjectMeta().GetNamespace())
+
+	kvp.Value = v2resource
+	return kvp, nil
+}
+
+func convertKVPairToResource(kvp *model.KVPair, resourceKind string) (Resource, error) {
+	var resource Resource
+	switch resourceKind {
+	case apiv2.KindBGPPeer:
```

Comment:
none of this switching should be necessary.  You already have the correct type in your hand